### PR TITLE
Added the user guide

### LINKED ISSUE #21: Create the documentation
Initially, the language container deployment documentation can be taken from the Transformers extension repo.

### PR DESCRIPTION
**Pull request by ahsimb**
closes #21

### CODE REVIEW

File: doc/user_guide/user-guide.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Python Extension Common User Guide
+
+## Language Container Deployer
+
+An extension would typically use UDF scripts to enable certain custom functionality within a database.
+In most cases, UDF scripts must be backed by a Language Container, that should be installed in the Exasol Database.
```
Comment (ckunki) [inline doc/user_guide/user-guide.md:6]:
```suggestion
In most cases, UDF scripts must be backed by a [Script Language Container (SLC)](https://github.com/exasol/script-languages-release/), that must be installed in the Exasol Database.
```

File: doc/user_guide/user-guide.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+A Script Language Container is a mechanism that allows the installation of the chosen programming language and
```
Comment (ckunki) [inline doc/user_guide/user-guide.md:7]:
```suggestion
SLCs allow the installation of the chosen programming language and
```

File: doc/user_guide/user-guide.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
+# Python Extension Common User Guide
+
+## Language Container Deployer
+
+An extension would typically use UDF scripts to enable certain custom functionality within a database.
```
Comment (ckunki) [inline doc/user_guide/user-guide.md:5]:
```suggestion
An extension would typically use [UDF scripts](https://docs.exasol.com/db/latest/database_concepts/udf_scripts.htm) to enable certain custom functionality within a database.
```

File: doc/user_guide/user-guide.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+In most cases, UDF scripts must be backed by a Language Container, that should be installed in the Exasol Database.
+A Script Language Container is a mechanism that allows the installation of the chosen programming language and
+necessary dependencies in the Exasol Database.
+
+The language container for a particular Extension gets downloaded and installed by executing a deployment script
+similar to the one below.
+
+  ```buildoutcfg
+  python -m an_exasol_extension.deploy language-container <options>
```
Comment (ckunki) [inline doc/user_guide/user-guide.md:14]:
```suggestion
  python -m <exasol_extension>.deploy language-container <options>
```

File: doc/user_guide/user-guide.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  ```
+
+The name of the script (```an_exasol_extension.deploy``` in the above command) can vary from one extension to another.
```
Comment (ckunki) [inline doc/user_guide/user-guide.md:17]:
```suggestion
The name of the script (```<exasol_extension>.deploy``` in the above command) can vary from one extension to another.
```

**Review by ckunki — CHANGES_REQUESTED** [review]

File: doc/user_guide/user-guide.md (hunk 2 above)
Comment (ahsimb) [reply doc/user_guide/user-guide.md:7]: An SLC allows ...

Comment (ckunki) [reply doc/user_guide/user-guide.md]: Yes, often singular is the better choice. Feel free to use your favorite phrasing.

**Review by ckunki — APPROVED** [review]

**Pull request merged**